puppersim/reacher/reacher_manual_control.py

```diff
@@ -0,0 +1,106 @@
+import reacher_kinematics
+import pybullet as p
+import puppersim.data as pd
+import time
+import math
+import numpy as np
+from absl import app
+from absl import flags
+import copy
+from pupper_hardware_interface import interface
+from serial.tools import list_ports
+
+flags.DEFINE_bool("run_on_robot", False, "Whether to run on robot or in simulation.")
+FLAGS = flags.FLAGS
+import pybullet_data
+
+KP = 10.0
+KD = 2.0
+MAX_CURRENT = 16.0
+
+HIP_OFFSET = 0.0335
+L1 = 0.08
+L2 = 0.11
+
+# def calculate_forward_kinematics_robot(joint_angles):
+#   # compute end effector pos in cartesian cords given angles
+
+#   x1 = L1 * math.sin(joint_angles[1])
+#   z1 = L1 * math.cos(joint_angles[1])
+
+#   x2 = L2 * math.sin(joint_angles[1] + joint_angles[2])
+#   z2 = L2 * math.cos(joint_angles[1] + joint_angles[2])
+
+#   foot_pos = np.array([[HIP_OFFSET],
+#                       [x1 + x2], 
+#                       [z1 + z2]
+#                       ])
+
+#   rot_mat = np.array([[math.cos(-joint_angles[0]), -math.sin(-joint_angles[0]), 0],
+#                       [math.sin(-joint_angles[0]), math.cos(-joint_angles[0]), 0],
+#                       [0, 0, 1]
+#                       ])  
+
+#   end_effector_pos = np.matmul(rot_mat, foot_pos)
+
+#   xyz = np.transpose(end_effector_pos)
+
+#   return xyz[0]
+
+def main(argv):
+  run_on_robot = FLAGS.run_on_robot
+  p.connect(p.GUI)
+  p.configureDebugVisualizer(p.COV_ENABLE_RGB_BUFFER_PREVIEW, 0)
+  p.configureDebugVisualizer(p.COV_ENABLE_DEPTH_BUFFER_PREVIEW, 0)
+  p.configureDebugVisualizer(p.COV_ENABLE_SEGMENTATION_MARK_PREVIEW, 0)
+  p.setAdditionalSearchPath(pybullet_data.getDataPath())
+  URDF_PATH = pd.getDataPath() + "/pupper_arm.urdf"
+  reacher = p.loadURDF(URDF_PATH, useFixedBase=True)
+
+  jointIds = []
+  paramIds = []
+
+  p.setPhysicsEngineParameter(numSolverIterations=10)
+  p.changeDynamics(reacher, -1, linearDamping=0, angularDamping=0)
+
+  for j in range(p.getNumJoints(reacher)):
+    p.changeDynamics(reacher, j, linearDamping=0, angularDamping=0)
+    info = p.getJointInfo(reacher, j)
+    #print(info)
+    jointName = info[1]
+    jointType = info[2]
+    if (jointType == p.JOINT_PRISMATIC or jointType == p.JOINT_REVOLUTE):
+      jointIds.append(j)
+      paramIds.append(p.addUserDebugParameter(jointName.decode("utf-8"), -4, 4, 0))
+
+  if run_on_robot:
+    serial_port = next(list_ports.grep("usbmodem")).device
```

Comment:
make the code detect if you're on linux or mac and then grep for the appropriate serial device (usbmodem or acm-something)